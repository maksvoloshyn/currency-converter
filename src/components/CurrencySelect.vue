<template>
    <select
        class="border px-3 py-2 rounded-lg bg-white"
        aria-label="Currency selector"
        :value="code"
        @change="onCurrencyChange($event.target.value)"
    >
        <option :value="null">Select currency</option>
        <option v-for="{name, code} in currencies" :key="code" :value="code">{{ code }} - {{ name }}</option>
    </select>
</template>

<script>
    export default {
        name: 'currency-select',

        props: {
            currencies: {
                type: Array,
                required: true,
            },

            code: {
                type: String,
                default: null,
            },
        },

        methods: {
            onCurrencyChange(value) {
                this.$emit('change', value);
            },
        },
    };
</script>
