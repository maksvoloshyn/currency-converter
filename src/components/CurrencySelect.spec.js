import {shallowMount} from '@vue/test-utils';
import CurrencySelect from '@/components/CurrencySelect';

const currenciesMock = [
    {code: 'UAH', name: 'Ukrainian hryvna'},
    {code: 'EUR', name: 'Euro'},
    {code: 'USD', name: 'United States Dollar'},
];

function factory() {
    return shallowMount(CurrencySelect, {
        propsData: {
            currencies: currenciesMock,
            code: 'EUR',
        },
    });
}

describe('CurrencySelect', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = factory();
    });

    it('renders correct template', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    describe('when select element emits `change` event', () => {
        it('emits `change` event with a given value', async () => {
            wrapper.find('[value="UAH"]').setSelected();

            expect(wrapper.emitted('change')[0][0]).toBe('UAH');
        });
    });
});
