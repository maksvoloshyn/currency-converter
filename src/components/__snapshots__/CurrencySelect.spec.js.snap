// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelect renders correct template 1`] = `
<select
  aria-label="Currency selector"
  class="border px-3 py-2 rounded-lg bg-white"
>
  <option
    value=""
  >
    Select currency
  </option>
   
  <option
    value="UAH"
  >
    UAH - Ukrainian hryvna
  </option>
  <option
    value="EUR"
  >
    EUR - Euro
  </option>
  <option
    value="USD"
  >
    USD - United States Dollar
  </option>
</select>
`;
